pkg/models/datasource_cache_test.go

```diff
@@ -163,7 +164,34 @@ func TestDataSourceCache(t *testing.T) {
 		So(err, ShouldBeNil)
 
 		Convey("Should skip TLS verification", func() {
-			So(tr.TLSClientConfig.InsecureSkipVerify, ShouldEqual, true)
+			So(tr.transport.TLSClientConfig.InsecureSkipVerify, ShouldEqual, true)
+		})
+	})
+
+	Convey("When caching a datasource proxy with custom headers specified", t, func() {
```

Comment:
Feels like we're missing a test of the `RoundTrip` which verifies that custom headers actually are set in the request. Could you change this test and call `ds.GetHttpTransport` instead of or in addition to getCustomHeaders` and make a test HTTP request using https://golang.org/pkg/net/http/httptest/?